README.md

```diff
@@ -1,33 +1,71 @@
-# hedera-smart-contract-starter
+# Upgradeable Smart Contracts with Proxies and Beacon Proxies
 
-This repository contains a smart contract starter project that utilizes a proxy pattern to enable upgradeable contracts on Hedera. The proxy pattern allows for the logic of a contract to be upgraded without the need to redeploy the contract or transfer its assets to a new contract address. This allows for easy maintenance and updates to the contract's functionality. The starter includes a basic smart contract `Bank.sol` which is the logic contract and instructions for deploying/upgrading it on the Hedera.
+This project provides a hands-on demonstration of using upgradeable smart contracts with OpenZeppelin's upgrades plugins in Hardhat. Specifically, it illustrates the use of both Proxies and Beacon Proxies.
 
+## Understanding Proxies
 
-### Set up
-1. Edit values for variables in `.env_sample` and save as `.env`
+A proxy is a smart contract that serves as an intermediary between the user and the logic contract. It forwards calls and data to the logic contract and then returns the results. The address of the logic contract can be updated, thus allowing changes to the contract's behavior over time without changing the contract's address.
 
-Try running some of the following tasks:
+## Understanding Beacon Proxies
 
-```shell
-npx hardhat help
-npx hardhat compile
-npx hardhat test
-npx hardhat run deployment/scripts/deploy.ts #Deploy logic and proxy contracts
-npx hardhat run deployment/scripts/upgradeProxy.ts #Upgrade to new implementation
-```
+A Beacon Proxy is a proxy that references a Beacon contract to determine its implementation. By updating the Beacon contract, all associated Beacon Proxies will point to the new implementation, allowing simultaneous upgrades of multiple contracts.
 
-### Upgrade Flow
-1. Deploy `Bank.sol` (v1 logic contract) and `TransparentUpgradeableProxy.sol` (proxy contract)
-```
-npx hardhat run deployment/scripts/deploy.ts
-```
+## How It Works
 
-2. Update `Bank.sol` (v2 logic contract)
+This project contains two contracts: `Node` and `NodeV2`.
 
-3. Upgrade deployment
-```
-npx hardhat run deployment/scripts/upgradeProxy.ts
-```
+- `Node` is a simple contract that enables storing and retrieving a Hedera Topic ID.
```

Comment:
nit: `Node` is a heavily used word so since this serves a specific purpose maybe `TopicNode` or something more descriptive to avoid the mixup of concepts and architecture entities